cmd/npmExecuteLint.go

```diff
@@ -0,0 +1,169 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/npm"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type lintUtils interface {
+	Glob(pattern string) (matches []string, err error)
+
+	getExecRunner() execRunner
+	getGeneralPurposeConfig(configURL string) error
+}
+
+type lintUtilsBundle struct {
+	*piperutils.Files
+	execRunner *command.Command
+	client     *piperhttp.Client
+}
+
+func newLintUtilsBundle() *lintUtilsBundle {
+	return &lintUtilsBundle{
+		Files:  &piperutils.Files{},
+		client: &piperhttp.Client{},
+	}
+}
+
+func (u *lintUtilsBundle) getExecRunner() execRunner {
+	if u.execRunner == nil {
+		u.execRunner = &command.Command{}
+		u.execRunner.Stdout(log.Writer())
+		u.execRunner.Stderr(log.Writer())
+	}
+	return u.execRunner
+}
+
+func (u *lintUtilsBundle) getGeneralPurposeConfig(configURL string) error {
+	response, err := u.client.SendRequest(http.MethodGet, configURL, nil, nil, nil)
```

Comment:
Do we really need to download such files from the internet?

Reply:
The idea was to download our general purpose configuration for ESLint from the jenkins-library Github repo, instead of loading it as library resource in the Groovy wrapper, to be able to run the step locally and be independent of Jenkins.